### ISSUE TITLE
Linking only gets URL if app is already running

### ISSUE THREAD

**Issue by stenrap** [facebook/react-native#11488]
### Description

I have successfully set up deep linking in my iOS app following the instructions [in the docs](https://facebook.github.io/react-native/docs/linking.html), and if I put a custom URL into Safari, my running app gets focus again and receives the URL as expected.

However, if I double-tap the home button and swipe up on my app (to close it), then put my custom URL into Safari, my app launches but it doesn't get the URL.

I started with this in my Navigator:

```javascript
componentDidMount() {
    Linking.addEventListener('url', (event) => this.handleLink(event.url));
}

componentWillUnmount() {
    Linking.removeEventListener('url', this.handleLink);
}

handleLink(url) {
    this._navigator.resetTo({name: Constants.EMAIL_LOGIN});
}

render() {
    return Navigator({
        initialRoute: {name: Constants.LAUNCH},
        ref: (component) => this._navigator = component,
        renderScene: (route, navigator) => {
            if (route.name === Constants.LAUNCH) {
                return React.createElement(LaunchView, {navigator: navigator});
            } else if (route.name === Constants.EMAIL_LOGIN) {
                return React.createElement(EmailLoginView, {navigator: navigator});
            }
        }
    });
}
```

When that didn't work, I thought maybe I needed to call `Linking.getInitialURL()`...

```javascript
componentDidMount() {
    let url = Linking.getInitialURL().then((url) => {
        if (url) {
            this.handleLink(url);
        } else {
            console.log('Opened without a url...');
        }
    }).catch((err) => console.log('An error occurred:', err));
    Linking.addEventListener('url', (event) => this.handleLink(event.url));
}
```

...but it doesn't make a difference. Am I doing anything wrong? How can I get the URL when my app is not already running?

### Reproduction

Obviously this cannot be reproduced on rnplay.org because it's not possible to add a custom URL scheme to the app's info.plist file.

### Additional Information

* React Native version: 0.37.0
* Platform: iOS
* Operating System: MacOS

**Comment by stenrap** [comment]
To follow up, I also got linking working on my Android device; however my app _never_ gets the URL, regardless of whether it was already running or not.

**Comment by stenrap** [comment]
Turns out this is a duplicate of #11191. I'm not using universal links, but I got linking fully working in iOS and Android with the code in that bug.

**Comment by rezaarifian** [comment]
hayy ,,
why Linking only gets URL if app is already running is closed ??
it is Solved

**Comment by rezaarifian** [comment]
on IOS, app can recieve url scheme if application on foreground, this system protection from Airdrop.

this say's Guide Ios 
"After transferring an eligible document to your app’s container, iOS launches your app (if needed) and calls the application:openURL:sourceApplication:annotation: method of its app delegate. If your app is in the foreground, you should use this method to open the file and display it to the user. If your app is in the background, you might decide only to note that the file is there so that you can open it later. Because files transferred via AirDrop are encrypted using data protection, you cannot open files unless the device is currently unlocked. "

reference : https://developer.apple.com/library/content/documentation/iPhone/Conceptual/iPhoneOSProgrammingGuide/Inter-AppCommunication/Inter-AppCommunication.html

**Comment by rezaarifian** [comment]
it's still doesn't get the URL if application terminated just open application on iOS device.
why close this problem, are you any solved for this ?

**Comment by cubabit** [comment]
Hi - I am having this problem too and not sure what you mean by **with the code in that bug** as the code you posted above seems to be the same as the solution in that bug?
